### ISSUE TITLE
Планируется ли поддержка других браузеров кроме Google Chrome?

### ISSUE THREAD

**Issue by Dominus77** [GHopperMSK/svgMap#11]
Не работает в Mozilla Firefox!

**Comment by np25071984** [comment]
Да. В моих планах есть поддержка всех броузеров, в т.ч. мобильных.

**Comment by Dominus77** [comment]
Тогда чем вас смущает удаление опции добавления не поддерживаемого в заголовке SVG, атрибута transform?

**Comment by np25071984** [comment]
1. Давай уже на Ты? Я сторонник свободного общения и вполне адекватно реагирую на "все говно, я знаю как правильно". 
2. По поводу вопроса - мне кажется это хороший вариант, только я не вижу реквеста.
3. Повторюсь по поводу реквестов - они слишком большие. Мне нравятся многие идеи (например тултип закрепленный), но я не могу принять его, т.к. запрос содержит еще кучу изменений. Если бы этот реквест был разбит на несколько мелких - было бы гораздо удобнее.
4. Если посмотреть историю коммитов, то мы с тобой говнокодеры - половина коммитов связана с расстановкой отступов и правкой опечаток. Это пустая трата ресурсов на фоне того, что не реализованы основные функции (например, до сих пор нет работает зум).

**Comment by Dominus77** [comment]
1. Давай на Ты, могу иногда по привычке перейти на вы, воспитание))
2. [Поддержка атрибута transform в заголовке тега svg](https://github.com/GHopperMSK/svgMap/pull/9)
3. Сейчас я делаю Pull-request`ы раздельно, создавая ветку от master, с тематическим названием исправления. Что бы не заливать тебе в master, может сделаешь отдельную ветку куда запуливать реквесты, а от туда уже будешь мержить с веткой master.
[Тултип закрепленный](https://github.com/GHopperMSK/svgMap/pull/10), в этом реквесте всего 2 коммита, по сути 1, всё остальное как у тебя в master. Коммиты можно посмотреть.
4. В дальнейшем эти отступы и опечатки вносят хаос, поэтому их нужно исправлять сразу, что бы потом, не дебажить и не искать ошибку из за опечатки, во всём коде. Я работаю в IDE PhpStorm, он сам заботится о пробелах и расстановках в коде согласно стандарту PSR.
По поводу зума, я еще им и не занимался)) Посмотрел так мельком и возник вопрос, ты не смотрел библиотеки которые занимаются этим функционалом? Почему бы не использовать готовое решение?

P.S.
У меня есть некоторое количество опубликованных и поддерживаемых мной виджетов, которыми с успехом пользуются, народ не жалуется)
[The PHP Package Repository](https://packagist.org/?q=dominus77&p=0)

**Comment by np25071984** [comment]
2. Так я же смержил его. Там конфликт был с моей веткой. Сейчас пересмотрел свой russia.php и понял, что там вообще беда с ним. Конвертор, видимо, непутевый попался. Перешел полностью на твои json-ны.
3. Сделаю ветку develop. По поводу титульного закрепленного мне не нравится твое решение тем, что там хардкод.Ты задаешь размер строго. Это не допустимо, т.к. там могут быть любые данные (карточка с менеджером, например). Если пользователь захочет поменять - пусть сам меняет через стили. Но хардкодить размер это точно не вариант.
Но идея хорошая (посмотрел, у большинства так реализовано). Внедрил в v0.3.0.
4. У меня тоже PhpStorm. Единственная проблема - когда сохраняешь многострочную переменную в двойных кавычках, то он лишние проблемы на конце строк не подбирает. Это причина частых конфликтов с коллегами, не знаю что у них там.
По поводу готового решения - можно подумать. Только мне не совсем понятно что оно даст. Как я уже говорил виджет из 10 строк лучше чем виджет из 100. Поэтому рассмотреть можно, но в целом я бы лучше перешел на чистый javascript. Функционал простой, написать можно все самим. Зато отвязавшись от чужих библиотек можно облегчить многим страдание )

**Comment by Dominus77** [comment]
> По поводу титульного закрепленного мне не нравится твое решение тем, что там хардкод.Ты задаешь размер строго. Это не допустимо, т.к. там могут быть любые данные (карточка с менеджером, например). Если пользователь захочет поменять - пусть сам меняет через стили. Но хардкодить размер это точно не вариант.

Поэтому была и добавлена опция в настройки виджета, в которой можно динамически задавать позицию. Начальная позиция задана по умолчанию, а пользователь уже решает оставить так или её изменить согласно его требованиям. Например генерировать эти значения в php и вставлять в предусмотренные настройки виджета. В таблице стилей значения как то тоже не удобно динамически менять. Только в самом шаблоне через $this->registerCss(), но это пахнет костылём.

Мой вариант вашего виджета находится тут: https://github.com/Dominus77/svgMap/tree/myversion 

> Зато отвязавшись от чужих библиотек можно облегчить многим страдание)

Скорее создать себе. Изобретать тот велосипед который уже изобретён.))

**Comment by np25071984** [comment]
> В таблице стилей значения как то тоже не удобно динамически менять. Только в самом шаблоне через $this->registerCss(), но это пахнет костылём.

Не вижу в этом смысла, т.к. мне проще css добавить в проект и получить новый тултип.
